### PR TITLE
Update template .gitignore rules

### PR DESCRIPTION
**Pull request by xqwzts**
This PR updates the .gitignore files that get created from the create/package/plugin templates:

- Adds generated ios files
- Adds .vscode where missing

### CODE REVIEW

**Comment by xqwzts** [comment]
Should plugin projects really have `pubspec.lock` in their .gitignore? It's my understanding that they should be checked in [especially when publishing to pub]

**Comment by DanTup** [comment]
Slightly related - I'm not sure we should be putting `.vscode` into ignore files: https://github.com/flutter/flutter/issues/15555

File: packages/flutter_tools/templates/create/.gitignore.tmpl [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
-.atom/
 .dart_tool/
+
+.atom/
 .idea
 .vscode/
+
 .packages
 .pub/
+packages
```
Comment (Hixie) [inline packages/flutter_tools/templates/create/.gitignore.tmpl]: We should remove `packages` now actually. It's for the old package system.

**Comment by Hixie** [comment]
I don't have a strong opinion on the .vscode thing, so long as we do the same thing for other IDEs.

LGTM modulo comments.

Please let me know when this is ready to land!

File: packages/flutter_tools/templates/create/.gitignore.tmpl (hunk 1 above)
Comment (xqwzts) [reply packages/flutter_tools/templates/create/.gitignore.tmpl]: Done.

**Comment by xqwzts** [comment]
Updated, PTAL.

**Comment by xqwzts** [comment]
Updated to remove the `.atom` `.vscode` and `.idea` ignore rules as discussed in https://github.com/flutter/flutter/issues/15555

File: packages/flutter_tools/templates/create/.gitignore.tmpl [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
-.idea
-.vscode/
+
 .packages
 .pub/
+
 build/
 ios/.generated/
-packages
+ios/Flutter/Generated.xcconfig
```
Comment (xster) [inline packages/flutter_tools/templates/create/.gitignore.tmpl]: these are already in templates/create/ios.tmpl/.gitignore

File: packages/flutter_tools/templates/plugin/.gitignore.tmpl [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
+
 .packages
 .pub/
+pubspec.lock
+
 build/
 ios/.generated/
-packages
-pubspec.lock
+ios/Flutter/Generated.xcconfig
```
Comment (xster) [inline packages/flutter_tools/templates/plugin/.gitignore.tmpl]: prefer adding them to the ios.tmpl subdirectory

File: packages/flutter_tools/templates/create/.gitignore.tmpl (hunk 2 above)
Comment (xqwzts) [reply packages/flutter_tools/templates/create/.gitignore.tmpl]: Thanks, I wasn't aware of that.

File: packages/flutter_tools/templates/plugin/.gitignore.tmpl (hunk 3 above)
Comment (xqwzts) [reply packages/flutter_tools/templates/plugin/.gitignore.tmpl]: Done.

**Comment by Hixie** [comment]
@xqwzts Is this ready to land?

**Comment by xqwzts** [comment]
It is! Sorry, I must have forgotten to comment when I committed.

**Comment by Hixie** [comment]
Great, I'll land this when the tree is green.

**Comment by Hixie** [comment]
Looks like tree _is_ green and flutter-build is just confused above. Landing.

**Comment by xqwzts** [comment]
Thanks!

**Pull request merged**